### DISCUSSION TITLE
HTTP Basic authentication

### DISCUSSION THREAD

**Discussion by silverleaf21** [miguelgrinberg/microdot#218] [Q&A] [answered]
Hi,

is it currently possible on micropython (microdot 2.0.5 via pypi) to setup Basic HTTP Authentication for a MicroDot-Sever?
a single user / password would be enough in my case.

If yes, how would I do it?

Thanks!

**Comment by miguelgrinberg** [comment, marked as answer]
Updated answer as of 2025: support for Basic and Token authentication is now available. There is also support for cookie-based logins.

Original answer: There is no high-level support for Basic auth in the current release. This is actually a work in progress right now, I'll be adding that soon: https://github.com/miguelgrinberg/microdot/pull/217

**Reply by silverleaf21** [reply]
Ok, thanks for the information!

**Reply by EpocDotFr** [reply]
Hello, any update on this topic? Looked at the PR, looks awesome!

**Reply by miguelgrinberg** [reply]
No updates, the change is stalled, but I do intend to go back to it and finish the work. The part that does HTTP auth is fairly robust, the cookie based login is the area that needs a bit of extra work.

**Reply by maxime-anoloc** [reply]
Wasn't even expecting a full cookie-based login system 😉

**Reply by EpocDotFr** [reply]
Thank you Miguel!
